### DISCUSSION TITLE
Feedbck

### DISCUSSION THREAD

**Discussion by AlexJacu** [jpcornil-git/HA-sma#15] [General]
HA-sma also compatible with WEBBOX-K1

Webbox Plant:  5 sensors
  sma_webbox_0_griegytdy
  sma_webbox_0_griegytot
  sma_webbox_0_gripwr
  sma_webbox_0_msg
  sma_webbox_0_OpStt

Generators:  17 sensors Attention! Some sensors are localized into local language!
  sma_webbox_1_pac
e.g. sma_webbox_1_seriennummer

**Comment by jpcornil-git** [comment]
Thanks for the feedback
=> everything working fine if I understood correctly ? (I'll close then issue and add K1 to the supported device then)
Cheers
jpc

**Comment by AlexJacu** [comment]
Yes, works fine.

But if I could reduce the interval from 30 sec to 5 sec. it would be fine, the webbox updates the the data every 5 sec.

Gruß

A. Jacubowsky

**Comment by jpcornil-git** [comment]
You can do that using thé scan_interval parameter (yaml or config flow).

Note that chapter "4.3 Query Interval" of the [Sunny Webbox RPC User Manual](https://github.com/jpcornil-git/HA-sma/blob/main/Sunny-Webbox-remote-procedure-call-User-manual-v1.4.pdf) specifies that the interval should not be less than 30s

**Comment by AlexJacu** [comment]
OK. But I know they are very conservative.

Additional, the webbox are using RS-485 connection with the inverters

Alexander Jacubowsky
Jacubowsky EDV

**Comment by jpcornil-git** [comment]
There is probably some margin but not sure why you would need faster update rate (30s is already 120 measurements/hour for each sensor and gives you very good insight/resolution already) ... but feel free to experiment with more aggressive scan_interval values :-)

For my records, what FW version is running on your webbox K1 ? 
Cheers,
jpc

**Reply by AlexJacu** [reply]
Version Firmware 1.53 
RS485-Kommunikation
Konverter sensors:
1	E-Total		1	kWh
2	Event-Cnt	751	
3	Fac		50,03	Hz
4	Fehler		-------	
5	h-On		1	h
6	h-Total		1	h
7	Iac-Ist		1,288	A
8	Ipv		1,219	A
9	Netz-Ein	1	
10	Pac		300	W
11	Phase		L3	
12	Riso		3000	kOhm
13	Seriennummer	xxxxxxx
14	Status		Mpp	
15	Uac		233	V
16	U-Fan		0	V
17	Upv-Ist		271	V
18	Upv-Soll	271	V
19	Zac		0,444	Ohm

**Reply by AlexJacu** [reply]
10 seconds was a bit to short, the updates took 12 to 14 sec. so i changed to 15 sec. update.
At 15 sec. Interval, it results in 17 on HA. So I have set it to 13 sec. and the result in HA is 15 sec., fine for me.

Hint: check the delay in update and reduce the updateinterval by that delay. eg. be delay ist 2-3 sec. so i set to 13 sec. and in HA the chart hast to 90% a 15 sec. stepping.

Please note, that my HomeAssistant is running on a virtual machine on an performance server (4 CPU and 6 GB RAM) !!! 
Connection with webbox is via LAN! and konverter are connected via RS485 cable !
Please be carful to reduce the 30 sec. limit, if HA is installed on a Rasberry or other slow devices !!

![30-seconds](https://github.com/jpcornil-git/HA-sma/assets/22580543/4c85f0e8-a1f5-408c-8a35-2638c8029476)
![15-seconds](https://github.com/jpcornil-git/HA-sma/assets/22580543/0206df94-3159-40ac-8b72-74989c910fac)
![10-seconds](https://github.com/jpcornil-git/HA-sma/assets/22580543/bbca4f4f-697a-4ea6-8ef0-80296569dac0)

**Comment by jpcornil-git** [comment]
Just added your K1/FW version to the list of supported device, thanks for reporting.
Wrt scan_interval, the critical element won't be HA but the webbox CPU probably (running other tasks like regular updates to sunny portal if enabled)
=> if you observe strange behaviors, try to increase scan_interval back to its minimum spec'ed value
Cheers,
jpc

**Reply by AlexJacu** [reply]
yep. the webbox is in the cabinet and is cooled together with the server

**Reply by AlexJacu** [reply]
One point ia have, in the night, when the konverters are offline i get an failure text (Not available), instead of 0 W (pac)

**Comment by jpcornil-git** [comment]
When the inverter disconnects itself from the grid a number of parameters becomes 'unavailable', this is expected.
Cheers,
jpc
